### ISSUE TITLE
Allow changing the DNS Nameservers after initial creation

### ISSUE THREAD

**Issue by cwrau** [kubernetes-sigs/cluster-api-provider-openstack#1603]
/kind feature

**Describe the solution you'd like**
Currently, when you need to change the DNS servers of a cluster, or rather, all clusters, you have to manually go into OpenStack, remove the old DNS servers, add the new DNS servers, disable the capo validation webhook, wait for reconciliation, reenable the webhook and roll all machines.

We'd like for capo to do these things. 

**Anything else you would like to add:**

**Comment by mdbooth** [comment]
This sounds like an interesting and desirable feature, but probably hard to implement. I may be wrong, but my guess is that none of our regular contributors are likely to implement this any time soon (read: ever).

However, if you or somebody you know would like to work on it I'd be happy to discuss how it could be implemented.

**Comment by jichenjc** [comment]
actually we have struggled this before when doing OCP enablement.. this sounds like a general question to update the DNS of deployment server (which pre-set DNS during deploymnet) process.. ? Anyway I agree it's a very special use case and if someone want to contribute it will be a plus but I don't think it's high priority anytime soon

**Comment by EmilienM** [comment]
Almost a year without real work done here, I wonder if this will really happen or if we could close this one.

**Comment by nikParasyr** [comment]
So we actually hit this issue a couple of months ago.
We had to (maybe the order is not 100% correct):
1. remove the old DNS servers on Openstack level, add the new DNS servers on openstack level
2. disable the capo validation webhook, 
3. update cluster definitions with the new dns_nameservers
4. wait for reconciliation, reenable the webhook
5. reroll new machines ( we did with `clusterctl alpha rollout restart -n test machinedeployment/ or kubeadmcontrolplane/` )
 
I think having capo fully doing the process above is quite complicated.
If capo can update and reconcile the dns_nameservers attribute on the subnet, it  will (in my understanding) remove steps 1,2,4 of the process above and it would make it much easier:
1. update dns_nameservers on cluster definition
2. roll out new machines

I think this is also a much easier change on capo side ( but not 100% sure )
